lib/src/Stepper/Stepper.tsx

```diff
@@ -7,52 +7,107 @@ import "./stepper.scss";
 export type StepperProps = JSX.IntrinsicElements["input"] & {
     /** Element label */
     label?: string;
+    /** maximum value for the element */
+    max: number;
+    /** minimum value for the element */
+    min: number;
     /** callback when element value is decreased */
-    onDecrease: (event: React.MouseEvent<HTMLButtonElement>) => void;
+    onDecrease: (event: React.MouseEvent<HTMLButtonElement> | React.KeyboardEvent<HTMLDivElement>) => void;
     /** callback when element value is increased */
-    onIncrease: (event: React.MouseEvent<HTMLButtonElement>) => void;
+    onIncrease: (event: React.MouseEvent<HTMLButtonElement> | React.KeyboardEvent<HTMLDivElement>) => void;
     /** Feedback indicator */
     indicator?: Indicator;
     /** Props for the wrapper element (div) */
     wrapperProps?: JSX.IntrinsicElements["div"];
+    /** configuration for accessiblity texts */
+    accessibilityConfigs?: {
+        decrementText: string;
+        incrementText: string;
+        currentValueText: string;
+    };
 };
 
 /** A stepper makes it easier to input values that are in a narrow range */
 export const Stepper: React.FC<StepperProps> = React.forwardRef(
-    ({ label, onDecrease, onIncrease, indicator, wrapperProps = {}, ...props }: StepperProps, ref: React.ForwardedRef<HTMLInputElement>) => {
+    (
+        {
+            label,
+            onDecrease,
+            onIncrease,
+            indicator,
+            wrapperProps = {},
+            accessibilityConfigs = { decrementText: "Decrease by 1", incrementText: "Increase by 1", currentValueText: "Current value is " },
+            ...props
+        }: StepperProps,
+        ref: React.ForwardedRef<HTMLInputElement>
+    ) => {
         const [id, setId] = React.useState<string>("");
+        const labelId: string = randomId("stepper-label-");
+
+        const isDecrementDisabled: boolean = props.disabled || props.value === props.min;
+
+        const onDecrement = (event: React.MouseEvent<HTMLButtonElement> | React.KeyboardEvent<HTMLDivElement>) => {
+            return isDecrementDisabled ? null : onDecrease(event);
+        };
+
+        const isIncrementDisabled: boolean = props.disabled || props.value === props.max;
+
+        const onIncrement = (event: React.MouseEvent<HTMLButtonElement> | React.KeyboardEvent<HTMLDivElement>) => {
+            return isIncrementDisabled ? null : onIncrease(event);
+        };
+
+        const onKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+            switch (event.key) {
+                case "ArrowDown":
+                    onDecrement(event);
+                    break;
+                case "ArrowLeft":
+                    onDecrement(event);
+                    break;
+                case "ArrowUp":
+                    onIncrement(event);
+                    break;
+                case "ArrowRight":
+                    onIncrement(event);
+                    break;
+            }
+        };
```

Comment:
should we add `event.preventDefault()` to trap the focus within the stepper component, otherwise clicking on arrow button seemed to move the focus away from the component

Reply:
on a side note, i think we can add the support for `Home` and `End` button pressed per the guideline https://www.w3.org/TR/wai-aria-practices/#keyboard-interaction-17

Reply:
added 👍